🏠  Docs onboarding overhaul

dvsrepo opened the pull request:
- [x] Home page
- [x] Quickstart
- [x] Hugging Face Spaces settings
- [x] Deploy with Docker

dvsrepo committed:
Home review

dvsrepo committed:
Add draft quickstart and new getting started sections

dvsrepo committed:
how to configure hf spaces

dvsrepo committed:
resolve conflicts

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks
for more information, see https://pre-commit.ci

dvsrepo committed:
docker compose guide

dvsrepo committed:
resolve conflicts

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks
for more information, see https://pre-commit.ci

dvsrepo committed:
minor improvements

dvsrepo committed:
Merge branch 'docs/onboarding_improvement' of github.com:argilla-io/argilla into docs/onboarding_improvement

jfcalvo committed:
improvement: fix some small details for docker-compose.yaml example

jfcalvo commented on examples/deployments/docker/docker-compose.yaml:
```diff
@@ -36,7 +35,7 @@ services:
       - "ES_JAVA_OPTS=-Xms512m -Xmx512m"
       - cluster.routing.allocation.disk.threshold_enabled=false
-      #- xpack.security.enabled=false
+      - xpack.security.enabled=false
```
@frascuchon I have disabled Elastic security so users can use the docker compose file without setting authentication for it. Tell me if this is something totally not recomendable to do.

jfcalvo commented on examples/deployments/docker/docker-compose.yaml:
```diff
@@ -12,7 +9,8 @@ services:
       ARGILLA_ELASTICSEARCH: http://elasticsearch:9200
       ARGILLA_AUTH_SECRET_KEY: ${ARGILLA_AUTH_SECRET_KEY:? Please generate a 32 character random string with `openssl rand -hex 32`}
```
@frascuchon we don't really need to set a value for `ARGILLA_AUTH_SECRET_KEY`, tell me if you think we can remove it and make it easy to start working with the docker compose without downloading it and modifying it.

jfcalvo replied:
Also I don't fully remember if some of the other environment variables present in this file are deprecated by the changes you made. Please can you take a look?

frascuchon replied:
Done https://github.com/argilla-io/argilla/pull/5315/commits/fc264a865c60f81e67a589c848b897573abd67ce

jfcalvo replied:
You didn't remove `ARGILLA_AUTH_SECRET_KEY` is that ok?

frascuchon replied:
No. We can remove yes. Also, docs are clear when using several workers, so it's fine.

jfcalvo commented on examples/deployments/docker/docker-compose.yaml:
```diff
@@ -1,7 +1,4 @@
-version: "3.9"
```
`version` is deprecated so I have removed it.

jfcalvo commented on examples/deployments/docker/docker-compose.yaml:
```diff
@@ -48,6 +47,7 @@ services:
     volumes:
       - elasticdata:/usr/share/elasticsearch/data/
+
   kibana:
```
@frascuchon is kibana really needed? Can we remove it?

dvsrepo replied:
I would vote for removing it yes

dvsrepo replied:
or at least comment it out, maybe better

frascuchon replied:
I prefer to remove it.

dvsrepo committed:
Merge branch 'docs/onboarding_improvement' of github.com:argilla-io/argilla into docs/onboarding_improvement

dvsrepo committed:
Improve quickstart and UI start page

frascuchon commented on argilla/docs/getting_started/how-to-configure-argilla-on-huggingface.md:
````diff
@@ -0,0 +1,128 @@
+
+To use the Python SDK with private Spaces you need to specify your `HF_TOKEN` which can be found [here](https://huggingface.co/settings/tokens), when creating the client:
+
+```python
+import argilla as rg
+
+HF_TOKEN = "..."
+
+client = rg.Argilla(
+    api_url="<api_url>",
+    api_key="<api_key>"
+    headers={"Authorization": f"Bearer {HF_TOKEN}"}
+)
+```
````
@dvsrepo I think we can update the start-page code snippet with this, and also include a link to the documentation

frascuchon commented on argilla/docs/getting_started/how-to-configure-argilla-on-huggingface.md:
```diff
@@ -0,0 +1,128 @@
+
+
+## Space Secrets overview
+There's two optional secrets to set up the `USERNAME` and `PASSWORD` of the `owner` of the Argilla Space. Remember that, by default Argilla Spaces are configured with a *Sign in with Hugging Face* button, which is also used to grant an `owner` to the creator of the Space.
```
```suggestion
There's two optional secrets to set up the `USERNAME` and `PASSWORD` of the `owner` of the Argilla Space. Remember that, by default Argilla Spaces are configured with a *Sign in with Hugging Face* button, which is also used to grant an `owner` to the creator of the Space for personal spaces.
```

jfcalvo replied:
```suggestion
## Space Secrets overview

There's two optional secrets to set up the `USERNAME` and `PASSWORD` of the `owner` of the Argilla Space. Remember that, by default Argilla Spaces are configured with a *Sign in with Hugging Face* button, which is also used to grant an `owner` to the creator of the Space.
```

frascuchon committed:
chore: Remove old env variables

frascuchon committed:
chore: Remove kibana

frascuchon committed:
Update docker-compose.yaml

Remove ARGILLA_AUTH_SECRET_KEY

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks
for more information, see https://pre-commit.ci

davidberenstein1957 commented on argilla/docs/index.md:
```diff
@@ -57,15 +58,15 @@ Gathering data is a time-consuming process. Argilla helps by providing a tool th
 
-- [Cleaned UltraFeedback dataset](https://huggingface.co/datasets/argilla/ultrafeedback-binarized-preferences-cleaned) used to fine-tune the [Notus](https://huggingface.co/argilla/notus-7b-v1) and [Notux](https://huggingface.co/argilla/notux-8x7b-v1) models. The original UltraFeedback dataset was curated using Argilla UI filters to find and report a bug in the original data generation code. Based ...
-- [distilabeled Intel Orca DPO dataset](https://huggingface.co/datasets/argilla/distilabel-intel-orca-dpo-pairs) used to fine-tune the [improved OpenHermes model](https://huggingface.co/argilla/distilabeled-OpenHermes-2.5-Mistral-7B). This dataset was built by combining human curation in Argilla with AI feedback from distilabel, leading to an improved version of the Intel Orca dataset and outperf ...
+- [cleaned UltraFeedback dataset](https://huggingface.co/datasets/argilla/ultrafeedback-binarized-preferences-cleaned) and the [Notus](https://huggingface.co/argilla/notus-7b-v1) and [Notux](https://huggingface.co/argilla/notux-8x7b-v1) models, where we improved benchmark and empirical human judgment for the Mistral and Mixtral models with cleaner data using **human feedback**.
+- [distilabeled Intel Orca DPO dataset](https://huggingface.co/datasets/argilla/distilabel-intel-orca-dpo-pairs) and the [improved OpenHermes model](https://huggingface.co/argilla/distilabeled-OpenHermes-2.5-Mistral-7B), show how we improve model performance by filtering out 50% of the original dataset through **human and AI feedback**.
```
@dvsrepo I believe I had already updated these last section based on the feedback on the HF hub PRs.

dvsrepo replied:
Yes, I'm not sure what happened but I brought those changes here, there shouldn't be conflicts now

davidberenstein1957 reviewed (commented):
Hi. I think we've made some changes to the docs in `main` that are not represented and might lead to merge conflicts. Among others, already having a Hugging Face Spaces deployment guide. This might lead to some wrong redirect, so we need to keep track of this.

davidberenstein1957 commented on argilla/docs/getting_started/how-to-configure-argilla-on-huggingface.md:
```diff
@@ -0,0 +1,128 @@
+!!! tip "Looking to get started easily?"
+    If you just discovered Argilla and want to get started quickly, go to the [Quickstart guide](quickstart.md).
+
+## Persistent storage
```
```suggestion
## Persistent storage

```

davidberenstein1957 commented on argilla/docs/getting_started/how-to-configure-argilla-on-huggingface.md:
```diff
@@ -0,0 +1,128 @@
+    - **Your Space will be restarted and existing data will be lost**. From now on, all the new data you create in Argilla will be kept safely
+    - **Recover your data**, by following the above mentioned guides.
+
+## How to configure and disable OAuth access
```
```suggestion
## How to configure and disable OAuth access

```

davidberenstein1957 commented on argilla/docs/getting_started/how-to-configure-argilla-on-huggingface.md:
````diff
@@ -0,0 +1,128 @@
+  - name: argilla
+  - name: community-initiative
+```
+
````
```suggestion
```

davidberenstein1957 commented on argilla/docs/getting_started/how-to-configure-argilla-on-huggingface.md:
```diff
@@ -0,0 +1,128 @@
+
+    For Argilla Spaces with many users, it's strongly recommended to **set persistent storage to `Small`**.
+
+## How to use Private Spaces
```
```suggestion
## How to use Private Spaces

```

davidberenstein1957 reviewed (commented):
I think we need to update the env var config for this PR in https://github.com/argilla-io/argilla/pull/5291

frascuchon committed:
Merge branch 'develop' into docs/onboarding_improvement

dvsrepo committed:
quickstart improvements

dvsrepo committed:
Merge branch 'docs/onboarding_improvement' of github.com:argilla-io/argilla into docs/onboarding_improvement

leiyre committed:
update styles

dvsrepo committed:
Merge branch 'docs/onboarding_improvement' of github.com:argilla-io/argilla into docs/onboarding_improvement

leiyre committed:
update admonition icon size

dvsrepo committed:
add embed template space

dvsrepo committed:
Merge branch 'docs/onboarding_improvement' of github.com:argilla-io/argilla into docs/onboarding_improvement

nataliaElv commented on argilla/docs/getting_started/how-to-configure-argilla-on-huggingface.md:
```diff
@@ -0,0 +1,128 @@
+**Spaces get restarted due to maintainance, inactivity, and every time you change your Spaces settings**. Persistent storage enables Argilla to save to disk your datasets and configurations across restarts.
+
+!!! warning "Ephimeral FREE persistent storage"
+    Not setting persistent storage to `Small` means that **you will loose your data when the Space restarts**.
```
Maybe add a reference about the persistent storage warning in the UI and how to disable it? https://docs.argilla.io/en/latest/getting_started/installation/configurations/server_configuration.html#hugging-face

leiyre committed:
scale iframe

leiyre committed:
update media screen min

frascuchon commented on argilla/docs/getting_started/how-to-deploy-argilla-with-docker.md:
````diff
@@ -0,0 +1,47 @@
+  ARGILLA_HOME_PATH: /var/lib/argilla
+  ARGILLA_ELASTICSEARCH: http://elasticsearch:9200
+  ARGILLA_AUTH_SECRET_KEY: <add_generated_key_here>
+```
````
Since we've removed the ARGILLA_AUTH_SECRET_KEY from the docker-compose.yaml, I think we can simplify this part and remove it. cc @jfcalvo 
```suggestion
```

jfcalvo committed:
docs: add REINDEX_DATASETS as example at docker-compose.yaml

jfcalvo committed:
docs: remove some not needed instructions about ARGILLA_AUTH_SECRET_KEY

frascuchon committed:
chore: Configure default credentials

frascuchon committed:
chore: Pointing to docker setup

frascuchon committed:
chore: Add missing env vars setup

frascuchon committed:
Docs/review docker readme (#5325)

# Description
<!-- Please include a summary of the changes and the related issue.
Please also include relevant motivation and context. List any
dependencies that are required for this change. -->

Closes #<issue_number>

**Type of change**
<!-- Please delete options that are not relevant. Remember to title the
PR according to the type of change -->

- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing
functionality to not work as expected)
- Refactor (change restructuring the codebase without changing
functionality)
- Improvement (change adding some improvement to an existing
functionality)
- Documentation update

**How Has This Been Tested**
<!-- Please add some reference about how your feature has been tested.
-->

**Checklist**
<!-- Please go over the list and make sure you've taken everything into
account -->

- I added relevant documentation
- I followed the style guidelines of this project
- I did a self-review of my code
- I made corresponding changes to the documentation
- I confirm My changes generate no new warnings
- I have added tests that prove my fix is effective or that my feature
works
- I have added relevant notes to the CHANGELOG.md file (See
https://keepachangelog.com/)

---------

Co-authored-by: Daniel Vila Suero <daniel@argilla.io>

frascuchon committed:
Merge branch 'develop' into docs/onboarding_improvement

frascuchon commented on argilla/docs/getting_started/how-to-configure-argilla-on-huggingface.md:
```diff
@@ -0,0 +1,128 @@
+    If you plan to **use the Argilla Space beyond testing**, it's highly recommended to **set persistent storage to `Small`**.
+
+If you just want to quickly test or use Argilla for a few hours with the risk of loosing your datasets, choose `Ephemeral FREE`. `Ephemeral FREE` means your datasets and configuration will not be saved to disk, when the Space is restarted your datasets, workspaces, and users will be lost.
+
```
```suggestion

If you want to disable the persistence storage warning, you can set the environment variable `ARGILLA_SHOW_HUGGINGFACE_SPACE_PERSISTENT_STORAGE_WARNING=false`

```

frascuchon committed:
Apply suggestions from code review

Co-authored-by: David Berenstein <david.m.berenstein@gmail.com>
Co-authored-by: José Francisco Calvo <jose@argilla.io>

dvsrepo committed:
Update argilla/docs/getting_started/how-to-configure-argilla-on-huggingface.md

Co-authored-by: Paco Aranda <frascuchon@gmail.com>

dvsrepo committed:
Update argilla/docs/getting_started/how-to-configure-argilla-on-huggingface.md

Co-authored-by: Paco Aranda <frascuchon@gmail.com>

frascuchon committed:
Merge branch 'develop' into docs/onboarding_improvement

frascuchon committed:
chore: Update credentials setuP

davidberenstein1957 committed:
Fix typo corrections

frascuchon committed:
Merge branch 'develop' into docs/onboarding_improvement

dvsrepo committed:
Merge branch 'releases/2.0.0' into docs/onboarding_improvement

frascuchon committed:
Merge branch 'releases/2.0.0' into docs/onboarding_improvement

frascuchon committed:
chore: Add curl alternative

frascuchon commented on argilla/docs/getting_started/how-to-deploy-argilla-with-docker.md:
````diff
@@ -0,0 +1,39 @@
+wget -O docker-compose.yaml https://raw.githubusercontent.com/argilla-io/argilla/main/examples/deployments/docker/docker-compose.yaml
+```
+
+or using curl:
````
Added this in case wget is not present

dvsrepo committed:
final review
